lib/src/container/mod.rs

```diff
@@ -112,6 +154,69 @@ impl TryFrom<&str> for ImageReference {
     }
 }
 
+impl TryFrom<&str> for SignatureSource {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        match value {
+            "ostree-image-signed" => Ok(Self::ContainerPolicy),
+            "ostree-unverified-image" => Ok(Self::ContainerPolicyAllowInsecure),
+            o => match o.strip_prefix("ostree-remote-image:") {
+                Some(rest) => Ok(Self::OstreeRemote(rest.to_string())),
+                _ => Err(anyhow!("Invalid signature source: {}", o)),
+            },
+        }
+    }
+}
+
+impl TryFrom<&str> for OstreeImageReference {
+    type Error = anyhow::Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        let mut parts = value.splitn(2, ':');
+        // Safety: Split always returns at least one value.
+        let first = parts.next().unwrap();
+        let second = parts
+            .next()
+            .ok_or_else(|| anyhow!("Missing ':' in {}", value))?;
+        let (sigverify, rest) = match first {
+            "ostree-image-signed" => (SignatureSource::ContainerPolicy, Cow::Borrowed(second)),
+            "ostree-unverified-image" => (
+                SignatureSource::ContainerPolicyAllowInsecure,
+                Cow::Borrowed(second),
+            ),
+            "ostree-remote-registry" => {
+                let mut subparts = second.splitn(2, ':');
+                // Safety: Split always returns at least one value.
+                let remote = subparts.next().unwrap();
+                let rest = subparts
+                    .next()
+                    .ok_or_else(|| anyhow!("Missing second ':' in {}", value))?;
+                (
+                    SignatureSource::OstreeRemote(remote.to_string()),
+                    Cow::Owned(format!("registry:{}", rest)),
```

Comment:
I think we may end up with a `registry:docker://foo` at this point, which I think could become problematic down the road. Also, it would be good to have a testcase covering that.